android/src/toga_android/widgets/button.py

```diff
@@ -24,30 +25,38 @@ def create(self):
         self.native = A_Button(self._native_activity)
         self.native.setOnClickListener(TogaOnClickListener(button_impl=self))
 
+    def get_text(self):
+        return str(self.native.getText())
+
     def set_text(self, text):
-        self.native.setText(self.interface.text)
+        self.native.setText(text)
 
     def set_enabled(self, value):
         self.native.setEnabled(value)
 
     def set_font(self, font):
-        if font:
-            self.native.setTextSize(TypedValue.COMPLEX_UNIT_SP, font._impl.get_size())
-            self.native.setTypeface(font._impl.get_typeface(), font._impl.get_style())
+        self.native.setTextSize(TypedValue.COMPLEX_UNIT_SP, font._impl.get_size())
+        self.native.setTypeface(font._impl.get_typeface(), font._impl.get_style())
 
     def set_on_press(self, handler):
         # No special handling required
         pass
 
     def set_color(self, value):
-        if value:
+        if value is None:
+            pass
+            # TODO: set to system foreground
```

Comment:
To save us from working out what the system foreground color is, how about making this method save the original color when it's called for the first time, so it can restore it later?